### PR TITLE
Move wheel testing logic to downstream repos

### PR DESCRIPTION
**Pull request by vyasr**
This PR is the testing companion to #97. It also introduces a standardized publish workflow that may be reused between pure and manylinux builds (see #89).

### CODE REVIEW

**Comment by vyasr** [comment]
I need to rename the extension from `.yml` to `.yaml`. I'll do that before merging this PR, but first waiting for CI to pass on all the dependent PRs.

File: .github/workflows/wheels-test.yaml
Comment (divyegala) [inline .github/workflows/wheels-test.yaml:98]:
```suggestion
  tests:
```
Similar to https://github.com/rapidsai/shared-action-workflows/blob/016436cbac2d92f0f0b4f78a63262117068a8aee/.github/workflows/conda-cpp-tests.yaml#L99

File: .github/workflows/wheels-test.yaml
Comment (divyegala) [inline .github/workflows/wheels-test.yaml:87]: Can we follow naming convention like so? https://github.com/rapidsai/shared-action-workflows/blob/016436cbac2d92f0f0b4f78a63262117068a8aee/.github/workflows/wheels-build.yaml#L80-L87

File: .github/workflows/wheels-test.yaml
Comment (divyegala) [inline .github/workflows/wheels-test.yaml:112]: These are not necessary since they are now setup in the container directly

File: .github/workflows/wheels-test.yaml
Comment (divyegala) [inline .github/workflows/wheels-test.yaml:99]: Naming convention similar to https://github.com/rapidsai/shared-action-workflows/blob/016436cbac2d92f0f0b4f78a63262117068a8aee/.github/workflows/wheels-build.yaml#L96

Perhaps also add `matrix.gpu` in there

**Review by divyegala — COMMENTED** [review]
Could you please also add `RAPIDS_ARTIFACT_DIR` to `wheels-build.yaml`? I forgot to add it

**Comment by vyasr** [comment]
> Could you please also add `RAPIDS_ARTIFACT_DIR` to `wheels-build.yaml`? I forgot to add it

Done in https://github.com/rapidsai/shared-action-workflows/pull/117

**Review by divyegala — APPROVED** [review]

File: .github/workflows/wheels-publish.yaml
Comment (divyegala) [inline .github/workflows/wheels-publish.yaml:46]: Should we change this image?

File: .github/workflows/wheels-test.yaml
Comment (divyegala) [inline .github/workflows/wheels-test.yaml:70]: These don't seem needed anymore

File: .github/workflows/wheels-test.yaml
Comment (divyegala) [inline .github/workflows/wheels-test.yaml:75]: Is this line necessary?

File: .github/workflows/wheels-publish.yaml
Comment (vyasr) [reply .github/workflows/wheels-publish.yaml:46]: Good eye. Switching to `rapidsai/ci-wheel:cuda12.0.1-centos7-py3.10`

**Pull request merged**
